code-challenges.test.js

```diff
@@ -15,37 +15,126 @@
+
+const shuffleArray = (array) => {
+  let removeFirst = array.shift()
+  for (let i = array.length - 1; i > 0; i--) {
+    let newArray = Math.floor(Math.random() * (i + 1))
+    let a = array[newArray]
+    let b = array[i]
+    array[i] = a
+    array[newArray] = b
+  }
+  return array
+}
+
+// PASS  ./code-challenges.test.js
+// shuffleArray
+//   ✓ returns a new array that has removed the first item from the original array and shuffled the rest of the items. (1 ms)
+
+// Test Suites: 1 passed, 1 total
+// Tests:       1 passed, 1 total
 
 // --------------------2) Create a function that takes in an object that contains up votes and down votes and returns the end tally.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const votes1 = { upVotes: 13, downVotes: 2 }
-// Expected output: 11
-const votes2 = { upVotes: 2, downVotes: 33 }
-// Expected output: -31
+describe("totalVotes", () => {
+  const votes1 = { upVotes: 13, downVotes: 2 }
+  // Expected output: 11
+  const votes2 = { upVotes: 2, downVotes: 33 }
+  //Expected output: -31
+  it("Function that takes in an object containing up and down votes exectues a mathmatical expression to find the end total.", () => {
+    expect(totalVotes(votes1)).toEqual(11)
+    expect(totalVotes(votes2)).toEqual(-31)
+  })
+})
 
 // b) Create the function that makes the test pass.
+//Puesdo C:
+//Create the functions named above(totalVotes).
+// Give the funciton one parameter called 'object' to match the data we are passing in.
+// To find the number of votes create a simple mathmatical formula of upvote - downvote. This will give the total.
+// Since the data is in an object we need to call on the key-value pair to retrieve the information.
+// This can be accomplied by dot notation.
+// Place the mathmatical formula in a variable called total, so we can just return the variable for cleanliness.
+// Breathe because your crushing it...
+
+const totalVotes = (object) => {
+  let uVotes = object.upVotes
+  let dVotes = object.downVotes
+  let total = uVotes - dVotes
+  return total
+}
 
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
 // a) Create a test with an expect statement using the variables provided.
+describe("noDuplicates", () => {
+  const dataArray1 = ["array", "object", "number", "string", "Boolean"]
+  const dataArray2 = ["string", "null", "Boolean", "string", "undefined"]
+  it("Function that takes in two arrays and returns one single array with no duplicate values from the original array.", () => {
+    expect(noDuplicates(dataArray1, dataArray2)).toEqual([
+      "array",
+      "object",
+      "number",
+      "string",
+      "Boolean",
+      "null",
+      "undefined",
+    ])
+  })
+})
 
 const dataArray1 = ["array", "object", "number", "string", "Boolean"]
+
```

Comment:
While it isn't currently breaking your code, I would encourage you to cut and paste the test variables inside the scope of the test rather than creating duplicates.